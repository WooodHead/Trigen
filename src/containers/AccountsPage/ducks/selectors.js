import { createSelector } from 'reselect';


// Simple input selectors
export const getAccounts = (state) => state.accounts.Accounts;
export const getAccountCount = (state) => state.accounts.Accounts.length;
export const getEditedAccounts = (state) => state.accounts.EditedAccounts;
export const getAccountFakeID = (_, props) => props.fakeID;
export const getAccountFilter = (state) => state.accounts.AccountFilter;


export const makeGetEditedAccount = () => {
  return createSelector(
    [ getEditedAccounts, getAccountFakeID ],
    ( EditedAccounts, fakeID ) => {
      return EditedAccounts[fakeID];
    }
  );
};

export const makeGetFilteredAccounts = () => {
  return createSelector(
    [ getAccounts, getAccountFilter ],
    ( accounts, filter ) => {
      const filteredAccounts = accounts.filter(account => {
        return (
          account.username.includes(filter) ||
          account.domain.includes(filter)
        );
      });

      return filteredAccounts;
    }
  );
};
