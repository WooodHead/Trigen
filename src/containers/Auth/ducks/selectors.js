export const getLoggedIn = state => state.auth.user.isLoggedIn;
export const getMaxAccounts = state => state.auth.user.data.maxAccounts;
