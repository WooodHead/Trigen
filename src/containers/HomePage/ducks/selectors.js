export const getPassword = state => state.PasswordReducer.Password;
