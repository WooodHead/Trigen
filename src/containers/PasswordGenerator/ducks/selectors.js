export const getPassword = state => state.PasswordReducer.Password;
export const getProgress = state => state.PasswordReducer.InProgress;
