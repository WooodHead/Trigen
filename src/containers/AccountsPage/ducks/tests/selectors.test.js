import * as selectors from '../selectors';


describe('Selectors', () => {
  const state = {
    accounts: {
      Accounts: [
        {
          fakeID: 'fakeID',
          id: 0,
          username: 'ttester',
          domain: 'example.com',
        }
      ],
      EditedAccounts: {
        'fakeID': {
          fakeID: 'fakeID',
          id: 0,
          username: 'dudeDudsen',
          domain: 'example.com',
        }
      },
      AccountFilter: 'exa',
      AccountListStatus: { isLoading: false },
      AccountStatus: { 'fakeID': { isLoading: 'false' }},
    }
  };

  describe('Simple input selectors', () => {
    it('should getAccounts from state', () => {
      expect(selectors.getAccounts(state)).toEqual(
        [
          {
            fakeID: 'fakeID',
            id: 0,
            username: 'ttester',
            domain: 'example.com',
          }
        ]
      );
    });

    it('should getEditedAccounts from state', () => {
      expect(selectors.getEditedAccounts(state)).toEqual({
        'fakeID': {
          fakeID: 'fakeID',
          id: 0,
          username: 'dudeDudsen',
          domain: 'example.com',
        }
      });
    });

    it('should getAccountFakeID from props', () => {
      expect(selectors.getAccountFakeID(state, { fakeID: 'fakeID' })).toEqual(
        'fakeID'
      );
    });

    it('should getAccountCount()', () => {
      expect(selectors.getAccountCount(state)).toBe(1);
    });

    it('should getAccountFilter()', () => {
      expect(selectors.getAccountFilter(state)).toBe('exa');
    });

    it('should getAccountListStatus()', () => {
      expect(selectors.getAccountListStatus(state)).toMatchSnapshot();
    });

    it('should getAccountStatus', () => {
      expect(selectors.getAccountStatus(state)).toMatchSnapshot();
    })
  });

  describe('Real selectors', () => {
    it('should make getEditedAccount and get the account', () => {
      const getEditedAccount = selectors.makeGetEditedAccount();

      expect(getEditedAccount(state, { fakeID: 'fakeID' })).toEqual({
        fakeID: 'fakeID',
        id: 0,
        username: 'dudeDudsen',
        domain: 'example.com',
      });
    });

    it('should make getFilteredAccounts', () => {
      const getFilteredAccounts = selectors.makeGetFilteredAccounts();

      expect(getFilteredAccounts(state)).toMatchSnapshot();
    });

    it('should make getAccountStatusByFakeId', () => {
      const getAccountStatusByFakeID = selectors.makeGetAccountStatusByFakeID();

      expect(getAccountStatusByFakeID(state, { fakeID: 'fakeID'})).toMatchSnapshot();
    });
  });
});
