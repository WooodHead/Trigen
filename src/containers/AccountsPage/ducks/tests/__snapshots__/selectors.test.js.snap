exports[`Selectors Real selectors should make getAccountStatusByFakeId 1`] = `
Object {
  "isLoading": "false",
}
`;

exports[`Selectors Real selectors should make getFilteredAccounts 1`] = `
Array [
  Object {
    "domain": "example.com",
    "fakeID": "fakeID",
    "id": 0,
    "username": "ttester",
  },
]
`;

exports[`Selectors Simple input selectors should getAccountListStatus() 1`] = `
Object {
  "isLoading": false,
}
`;

exports[`Selectors Simple input selectors should getAccountStatus 1`] = `
Object {
  "fakeID": Object {
    "isLoading": "false",
  },
}
`;
